Contact/Search.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Nate.ContactApp
+{
+    public static class Search
+    {
+        private static string lastSearchString = string.Empty;
+        private static List<Contact> lastSearch;
+
+        public static List<Contact> Filter(List<Contact> ContactList, string parameter)
+        {
+
+            var results = new List<Contact>();
+
+            if (parameter.Contains(lastSearchString) && lastSearchString != string.Empty)
+                if (parameter == lastSearchString)
+                    results = lastSearch;
+                else
+                    results = searchAllFields(lastSearch, parameter);
+            else
+                results = searchAllFields(ContactList, parameter);
+
+            lastSearch = results;
+            lastSearchString = parameter;
+
+            return results;
+        }
+
+        private static List<Contact> searchAllFields(List<Contact> database, string parameter)
+        {
+            var searchData =
+            from contact in database
+            where contact.IsActiveRecord && (
+            contact.FirstName.Contains(parameter) ||
```

Comment:
I believe these are gong to be case sensitive compares which you don't typically want.

Reply:
You're right. I will need to update this to be case-agnostic.

Reply:
This has now been updated. I borrowed some code for an extension overload of Contains that I liked and would allow for a fairly simple implementation.